eli5/sklearn/permutation_importance.py

```diff
@@ -214,8 +214,13 @@ def _cv_scores_importances(self, X, y, groups=None, **fit_params):
         cv = check_cv(self.cv, y, is_classifier(self.estimator))
         feature_importances = []  # type: List
         base_scores = []  # type: List[float]
+        weights = fit_params.get('sample_weight', None)
+        fit_params.pop('sample_weight', None)
         for train, test in cv.split(X, y, groups):
-            est = clone(self.estimator).fit(X[train], y[train], **fit_params)
+            if weights is None:
+                est = clone(self.estimator).fit(X[train], y[train], **fit_params)
+            else:
+                est = clone(self.estimator).fit(X[train], y[train], sample_weight=weights[train], **fit_params)
```

Comment:
This does the right thing, but it's possible to improve this a bit - instead of repeating almost the same line twice, it's possible to modify ``fit_params`` or create new ``fold_fit_params``, putting modified ``sample_weight`` there - then the same call can be done for both cases, avoiding repetition.

Reply:
have a look at the new changes and see if they make sense. Good suggestions